sdk/python/kfp/_client.py

```diff
@@ -295,60 +296,143 @@ def list_pipelines(self, page_token='', page_size=10, sort_by=''):
     """
     return self._pipelines_api.list_pipelines(page_token=page_token, page_size=page_size, sort_by=sort_by)
 
+
   # TODO: provide default namespace, similar to kubectl default namespaces.
-  def run_pipeline(self, experiment_id, job_name, pipeline_package_path=None, params={}, pipeline_id=None, namespace=None):
-    """Run a specified pipeline.
+  def _create_job(self, experiment_id, job_name, start_time, end_time, interval_second, max_concurrency, params, pipeline_package_path, pipeline_id, namespace, enabled=None):
```

Comment:
i think this function should be split into multiple functions and compose together for recurring vs non recurring instead of a giant function that does everything

Reply:
also are there tests somewhere we can add to?

Reply:
I haven't found unit tests, only integration tests (e2e). Can we anyway merge this PR now, so that I can test if it works properly with `teiresias`.

I can add tests in a separate PR if necessary.